docs/project/release-policy.md

```diff
@@ -0,0 +1,43 @@
+# Release Policy
+
+Crystal releases have a version indicated by a major, minor and patch number.
+
+The current major branch number is `1`.
+
+New features are added in minor releases (`1.x.0`) which are regularly scheduled every three months.
+
+Patch releases contain only important bug fixes and are released when necessary.
+They usually only appear for the latest minor branch.
+
+New releases are announced at [crystal-lang.org/releases](https://crystal-lang.org/releases) ([RSS feed](https://crystal-lang.org/releases)).
+
+There are currently no plans for a new major release.
+
+## Backwards compatibility
+
+Minor and patch releases are backwards compatible: Well-defined behaviours and documented APIs in a given version
```

Comment:
How about:

```suggestion
Minor and patch releases are backwards compatible: any use of documented API from the standard library is expected to work in future versions within the same major branch. The same is expected for undocumented but not hidden API (that is, public and without `:nodoc:`) and any language construct. There are no guarantees over any other aspect, like compiler internals or `:nodoc:` API.
```

Reply:
This feels a bit too wordy and thus hard to grasp. I'll try to rephrase this with fewer words.